From a7c968c9fe595811fa4e251c0690991c5c6e5d06 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:48:36 -0400
Subject: [iommu] vt-d: Make iommu_dummy() take struct device instead of struct pci_dev

Message-id: <20140905014836.31297.60830.stgit@gir.stowe>
Patchwork-id: 91321
O-Subject: [RHEL7.1 PATCH 28/65] iommu/vt-d: Make iommu_dummy() take struct device instead of struct pci_dev
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 3d89194a94da0c49f9d21720df6bbaf7c848449c

commit 3d89194a94da0c49f9d21720df6bbaf7c848449c
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Thu Mar 6 15:59:26 2014 +0000

    iommu/vt-d: Make iommu_dummy() take struct device instead of struct pci_dev

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index ace088e..2079cb6 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2882,9 +2882,9 @@ static inline struct dmar_domain *get_valid_domain_for_dev(struct pci_dev *dev)
  return __get_valid_domain_for_dev(dev);
 }
 
-static int iommu_dummy(struct pci_dev *pdev)
+static int iommu_dummy(struct device *dev)
 {
- return pdev->dev.archdata.iommu == DUMMY_DEVICE_DOMAIN_INFO;
+ return dev->archdata.iommu == DUMMY_DEVICE_DOMAIN_INFO;
 }
 
 /* Check if the pdev needs to go through non-identity map and unmap process.*/
@@ -2896,13 +2896,13 @@ static int iommu_no_mapping(struct device *dev)
  if (unlikely(!dev_is_pci(dev)))
   return 1;
 
- pdev = to_pci_dev(dev);
- if (iommu_dummy(pdev))
+ if (iommu_dummy(dev))
   return 1;
 
  if (!iommu_identity_mapping)
   return 0;
 
+ pdev = to_pci_dev(dev);
  found = identity_mapping(pdev);
  if (found) {
   if (iommu_should_identity_map(pdev, 0))
@@ -3801,7 +3801,7 @@ static int device_notifier(struct notifier_block *nb,
  struct pci_dev *pdev = to_pci_dev(dev);
  struct dmar_domain *domain;
 
- if (iommu_dummy(pdev))
+ if (iommu_dummy(dev))
   return 0;
 
  if (action != BUS_NOTIFY_UNBOUND_DRIVER &&
-- 
1.7.1